javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/internal/ConfigPropertiesTranslator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.extension.internal;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@SuppressWarnings("NonApiType")
+class ConfigPropertiesTranslator {
```

Comment:
This class is weird. It is named like it had a single purpose, that is converting property names to declarative config property names, but really also provides overrides for property values. I'd consider deleting it and moving the functionality into `DeclarativeConfigPropertiesBridge`.